server.rb

```diff
@@ -0,0 +1,189 @@
+require "sinatra/base"
+require "sqlite3"
+require "nokogiri"
+require 'active_support'
+require 'json'
+require 'securerandom'
+
+class Server < Sinatra::Base
+
+  # {
+  #  to: <UUID>
+  #  from: <UUID>
+  #  message: <String>
+  # }
+
+  post '/messages/:format?' do
+    case params['format']
+    when 'json' then
+      message = JSONMessage.new(request.body.read)
+      if message.valid?
+        id = message.save
+
+        message.send_message
+        [200, {message: {id: id}}.to_json]
+      else
+        [400]
+      end
+    when 'xml'
+      message = XMLMessage.new(request.body.read)
+      if message.valid?
+        id = message.save
+        message.send_message
+        builder = Nokogiri::XML::Builder.new do |xml|
+          xml.root do
+            xml.message do
+              xml.id id
+            end
+          end
+        end
+        [200, builder.to_xml]
+      else
+        [400]
+      end
+    else
+      [400]
+    end
+  end
+
+  get '/messages/:id/status/:format?' do
+    status = db.execute("SELECT status from messages WHERE id = '#{params['id']}' LIMIT 1")
```

Comment:
CAUTION!: This is going to create an opportunity for [SQL Injection](https://cheatsheetseries.owasp.org/cheatsheets/SQL_Injection_Prevention_Cheat_Sheet.html)
In this scenario using paramerized queries instead of string interpolation will mitigate the problem.